File: backend/src/database/seeders/AgreementSeeder.php
<?php

namespace Database\Seeders;

use Illuminate\Database\Seeder;
use App\Models\Agreement;

class AgreementSeeder extends Seeder
{
    /**
     * Run the database seeds.
     */
    public function run(): void
    {
        $agreements = [
            [
                'code' => 'convenio_multilateral',
                'name' => 'Convenio Multilateral',
                'description' => 'Convenio Multilateral del Impuesto sobre los Ingresos Brutos',
                'is_active' => true
            ],
        ];

        foreach ($agreements as $agreement) {
            Agreement::firstOrCreate(
                ['code' => $agreement['code']],
                $agreement
            );
        }
    }
}

File: backend-dotnet/src/Ikhana.Infrastructure/Data/Seeders/AgreementSeeder.cs
using Ikhana.Application.Common.Interfaces;
using Ikhana.Domain.Entities;
using Ikhana.Infrastructure.Persistence;
using Microsoft.EntityFrameworkCore;

namespace Ikhana.Infrastructure.Data.Seeders;

public class AgreementSeeder : IDataSeeder
{
    private readonly AppDbContext _context;

    public int Order => 3;

    public AgreementSeeder(AppDbContext context)
    {
        _context = context;
    }

    public async Task SeedAsync(CancellationToken cancellationToken = default)
    {
        if (await _context.Set<Agreement>().AnyAsync(cancellationToken))
            return;

        _context.Set<Agreement>().Add(new Agreement
        {
            Code = "convenio_multilateral",
            Name = "Convenio Multilateral",
            Description = "Convenio Multilateral del Impuesto sobre los Ingresos Brutos",
            IsActive = true,
        });

        await _context.SaveChangesAsync(cancellationToken);
    }
}
